### PR TITLE
feat(ui): add transitions

### LINKED ISSUE #151: Add smooth transitions
Add transitions when changing the chain and also when opening the feed details.

### PR DESCRIPTION
**Pull request by gabaldon**
close #151

### CODE REVIEW

File: packages/ui/static/sw.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+self.addEventListener('activate', function (e) {
+  self.registration
+    .unregister()
+    .then(function () {
+      return self.clients.matchAll()
+    })
+    .then(function (clients) {
+      clients.forEach((client) => client.navigate(client.url))
+    })
+})
```
Comment (Tommytrg) [inline packages/ui/static/sw.js:18]: Could you remove this file?
```suggestion
```

File: packages/ui/layouts/default.vue [hunk 2]
```diff
@@ -60,14 +67,25 @@ html {
...
+.background {
+  background-color: var(--bg);
+  transition: all 0.3s ease;
+  &.preload {
+    transition: none !important;
+    -webkit-transition: none !important;
+    -moz-transition: none !important;
+    -ms-transition: none !important;
+    -o-transition: none !important;
+  }
```
Comment (Tommytrg) [inline packages/ui/layouts/default.vue:82]: I can't see any change even when I force `preload` to be always `true`

File: packages/ui/components/footer/Footer.vue [hunk 3]
```diff
@@ -93,6 +93,10 @@ export default {
   font-size: var(--text-size-medium);
   color: var(--white-text);
   padding: 4px 0 4px 0;
+  transition: all 0.3 ease-in-out;
```
Comment (Tommytrg) [inline packages/ui/components/footer/Footer.vue:96]: I can't see any animation here

File: packages/ui/components/cards/UpcomingChainCard.vue [hunk 4]
```diff
@@ -1,12 +1,14 @@
 <template>
-  <a :href="path" target="_blank">
-    <div class="card-container">
-      <div class="title">
-        <SvgIcon class="img" :name="img" />
-        <p class="name title">{{ name }}</p>
+  <BaseCard>
```
Comment (Tommytrg) [inline packages/ui/components/cards/UpcomingChainCard.vue:2]: Do the upcoming chains need to be Card? Do they have border, background or any other Card property?

File: packages/ui/components/chart/ChartTooltip.vue [hunk 5]
```diff
@@ -84,6 +84,7 @@ export default {
   font-weight: bold;
   font-size: var(--text-size-title);
   background-color: var(--bg);
+  transition: background-color 0.3s ease;
```
Comment (Tommytrg) [inline packages/ui/components/chart/ChartTooltip.vue:87]: How could I see this animation?

File: packages/ui/components/NavBar.vue [hunk 6]
```diff
@@ -125,17 +125,7 @@ export default {
   align-items: center;
   background-color: var(--bg);
   height: 100px;
-  button,
```
Comment (Tommytrg) [inline packages/ui/components/NavBar.vue:128]: Is removing this related with this error?
image

File: packages/ui/components/NetworkOptions.vue [hunk 7]
```diff
@@ -52,24 +58,61 @@ export default {
     isSelected(option) {
       return option.toLowerCase() === this.selectedOption.toLowerCase()
     },
+    setTransition() {
+      this.transition = 'btn'
+    },
```
Comment (Tommytrg) [inline packages/ui/components/NetworkOptions.vue:63]: Where is this being used?

File: packages/ui/components/NetworkOptions.vue [hunk 8]
```diff
@@ -39,6 +40,11 @@ export default {
       required: true,
     },
   },
+  data() {
+    return {
+      transition: null,
+    }
+  },
```
Comment (Tommytrg) [inline packages/ui/components/NetworkOptions.vue:47]: Where are we using this?

File: packages/ui/components/Socials.vue [hunk 9]
```diff
@@ -32,11 +32,15 @@ export default {
...
   flex-wrap: wrap;
   margin-bottom: 24px;
+  transition: all 0.3 ease-in-out;
   .link {
     margin-right: 24px;
     &:last-of-type {
       margin-right: 0;
     }
+    &:hover {
+      transform: scale(1.1);
```
Comment (Tommytrg) [inline packages/ui/components/Socials.vue:42]:
```suggestion
      transform: scale(1.3);
```

File: packages/ui/components/SvgIcon.vue [hunk 10]
```diff
@@ -27,6 +27,7 @@ export default {
 .icon-size {
   width: 24px;
   height: 30px;
+  transition: all 0.3s ease;
```
Comment (Tommytrg) [inline packages/ui/components/SvgIcon.vue:30]: How can I see this animation?

File: packages/ui/components/ThemeSwitch.vue [hunk 11]
```diff
@@ -53,6 +53,9 @@ export default {
   font-size: 20px;
   width: 20px;
   height: 20px;
+  &:hover {
```
Comment (Tommytrg) [inline packages/ui/components/ThemeSwitch.vue:56]: If we are making bigger the icon, I'd also make bigger the button where the icon is

File: packages/ui/components/Transaction.vue [hunk 12]
```diff
@@ -74,7 +74,11 @@ export default {
 }
 .link {
   color: var(--witnet-transaction);
-  font-family: 'Roboto Mono', monospace;
+  font-family: Roboto Mono, monospace;
+  transition: all 0.3 ease-in-out;
```
Comment (Tommytrg) [inline packages/ui/components/Transaction.vue:78]: How can I see this animation?

File: packages/ui/components/breadcrumbs/BreadCrumbs.vue [hunk 13]
```diff
@@ -69,6 +71,20 @@ export default {
...
+  transform: translateX(-4px);
+}
+.slide-in-enter-to {
+  opacity: 1;
+  transform: translateY(0);
+}
+.slide-in-leave-to {
+  opacity: 0;
+  transform: translateX(-4px);
+}
```
Comment (Tommytrg) [inline packages/ui/components/breadcrumbs/BreadCrumbs.vue:87]: How could I see this transition?

File: packages/ui/components/cards/FeedCard.vue [hunk 14]
```diff
@@ -104,6 +106,17 @@ export default {
...
+  &.operational {
+    border: var(--card-border);
+  }
+  &.delay {
+    border: var(--delay-status-border);
+  }
+  &.error {
+    border: var(--error-status-border);
+  }
+}
```
Comment (Tommytrg) [inline packages/ui/components/cards/FeedCard.vue:119]: Is this being used?

File: packages/ui/components/cards/FeedCard.vue [hunk 15]
```diff
@@ -112,23 +125,10 @@ a {
...
-  &.operational {
-    border: var(--card-border);
-  }
-  &.delay {
-    border: var(--delay-status-border);
-  }
-  &.error {
-    border: var(--error-status-border);
-  }
+  transition: box-shadow 0.3s;
```
Comment (Tommytrg) [inline packages/ui/components/cards/FeedCard.vue:131]: How could I see this?

File: packages/ui/components/breadcrumbs/BreadCrumbs.vue [hunk 16]
```diff
@@ -85,6 +101,10 @@ export default {
     font-weight: bold;
     font-size: var(--text-size-medium);
     margin-right: 4px;
+    transition: all 0.3 ease-in-out;
```
Comment (Tommytrg) [inline packages/ui/components/breadcrumbs/BreadCrumbs.vue:104]: How could I see this?

File: packages/ui/components/SvgIcon.vue (hunk 10 above)
Comment (gabaldon) [reply packages/ui/components/SvgIcon.vue:30]: changing the theme

File: packages/ui/components/cards/FeedCard.vue (hunk 14 above)
Comment (gabaldon) [reply packages/ui/components/cards/FeedCard.vue:119]: Yes, when there is a delay the border is yellow

File: packages/ui/layouts/default.vue (hunk 2 above)
Comment (gabaldon) [reply packages/ui/layouts/default.vue:82]: The difference is when changing the theme  in the navbar background color

**Review by Tommytrg — APPROVED** [review]

File: packages/ui/static/sw.js (hunk 1 above)
Comment (Tommytrg) [reply packages/ui/static/sw.js:18]: @gabaldon I you remove this file, LGTM

**Review by Tommytrg — CHANGES_REQUESTED** [review]

**Review by Tommytrg — APPROVED** [review]

**Pull request merged**